diff --git a/fetchcommandwrapper/__main__.py b/fetchcommandwrapper/__main__.py
index 956d944..8c21bf4 100755
--- a/fetchcommandwrapper/__main__.py
+++ b/fetchcommandwrapper/__main__.py
@@ -4,6 +4,7 @@
 # Licensed under GPL v3 or later
 
 import os
+import re
 import sys
 from signal import SIGINT
 from textwrap import dedent
@@ -61,19 +62,31 @@ def parse_parameters():
         dest="link_speed_bytes",
         help="specify link speed (bytes per second). enables dropping of slow connections.",
     )
+    parser.add_argument(
+        "--github-proxy",
+        metavar="PROXY_URL",
+        dest="github_proxy",
+        help="use specified proxy for GitHub URLs (e.g., http://proxy.example.com:8080)",
+    )
     parser.add_argument("uri", metavar="URI", help=argparse.SUPPRESS)
     parser.add_argument("distdir", metavar="DISTDIR", help=argparse.SUPPRESS)
     parser.add_argument("file_basename", metavar="FILE", help=argparse.SUPPRESS)
 
     opts, opts.argv_extra = parser.parse_known_args()
+    print(opts)
 
     opts.distdir = opts.distdir.rstrip("/")
+    print(opts.distdir)
 
     opts.file_fullpath = os.path.join(opts.distdir, opts.file_basename)
 
     return opts
 
 
+def is_github_uri(uri):
+    return uri.startswith("https://github.com") or uri.startswith("https://raw.githubusercontent.com")
+
+
 def invoke_wget(opts):
     args = ["/usr/bin/wget", "-O", opts.file_fullpath]
     args.append("--tries=5")
@@ -134,8 +147,18 @@ def print_mirror_details(supported_mirror_uris):
             file=sys.stderr,
         )
 
+def replace_github_url(url,repUrl=None):
+    # prefix = repUrl if repUrl else 'https://gh-proxy.com'
+    if repUrl is None or repUrl == "" or (not is_github_uri(url)):
+        return url
+
+    return re.sub(
+        r'^https://(raw\.githubusercontent.com|github\.com)/',
+        rf'{repUrl.rstrip('/')}/\1/',
+        url
+    )
 
-def make_final_uris(uri, supported_mirror_uris):
+def make_final_uris(uri, supported_mirror_uris, github_proxy=None):
     final_uris = [
         uri,
     ]
@@ -162,6 +185,9 @@ def make_final_uris(uri, supported_mirror_uris):
                 random.shuffle(final_uris)
                 break
 
+    if not mirrors_involved:
+       final_uris = [replace_github_url(url, github_proxy) for url in final_uris]
+       print(f'Use Original URI: final_uris-> {final_uris}')
     return final_uris, mirrors_involved
 
 
@@ -194,9 +220,9 @@ def invoke_aria2(opts, final_uris):
     args.append("--allow-overwrite=true")
     args.append("--max-tries=5")
     args.append("--max-file-not-found=2")
-    args.append("--user-agent=Wget/1.19.1")
-    args.append("--split=%d" % wanted_connections)
-    args.append("--max-connection-per-server=1")
+    args.append('--user-agent="Mozilla/5.0 (X11; Linux x86_64; rv:148.0) Gecko/20100101 Firefox/148.0"')
+    args.append("--split=%d" % 16)
+    args.append("--max-connection-per-server=16")
     args.append("--uri-selector=inorder")
     args.append("--follow-metalink=mem")  # e.g. GNOME servers with mirrorbrain
     args.extend(opts.argv_extra)
@@ -229,7 +255,7 @@ def _inner_main():
 
     supported_mirror_uris = [e for e in gentoo_mirrors() if supported(e)]
 
-    final_uris, mirrors_involved = make_final_uris(opts.uri, supported_mirror_uris)
+    final_uris, mirrors_involved = make_final_uris(opts.uri, supported_mirror_uris, opts.github_proxy)
 
     if VERBOSE:
         print_greeting()
